### Importación de datos



In [2]:
import pandas as pd

url = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_1%20.csv"
url2 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_2.csv"
url3 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_3.csv"
url4 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_4.csv"

tienda = pd.read_csv(url)
tienda2 = pd.read_csv(url2)
tienda3 = pd.read_csv(url3)
tienda4 = pd.read_csv(url4)

tienda.head()

,Producto,Categoría del Producto,Precio,Costo de envío,Fecha de Compra,Vendedor,Lugar de Compra,Calificación,Método de pago,Cantidad de cuotas,lat,lon
0,Asistente virtual,Electrónicos,164300.0,6900.0,16/01/2021,Pedro Gomez,Bogotá,4,Tarjeta de crédito,8,4.60971,-74.08175
1,Mesa de comedor,Muebles,192300.0,8400.0,18/05/2022,Beatriz Morales,Medellín,1,Tarjeta de crédito,4,6.25184,-75.56359
2,Juego de mesa,Juguetes,209600.0,15900.0,15/03/2021,Juan Fernandez,Cartagena,1,Tarjeta de crédito,1,10.39972,-75.51444
3,Microondas,Electrodomésticos,757500.0,41000.0,03/05/2022,Juan Fernandez,Cali,4,Nequi,1,3.43722,-76.52250
4,Silla de oficina,Muebles,335200.0,20200.0,07/11/2020,Maria Alfonso,Medellín,5,Nequi,1,6.25184,-75.56359


#1. Análisis de facturación



In [8]:
# Sumamos el total de ventas de cada tienda
fact_1 = tienda['Precio'].sum()
fact_2 = tienda2['Precio'].sum()
fact_3 = tienda3['Precio'].sum()
fact_4 = tienda4['Precio'].sum()

print(f"Facturación Tienda 1: ${fact_1:,.2f}")
print(f"Facturación Tienda 2: ${fact_2:,.2f}")
print(f"Facturación Tienda 3: ${fact_3:,.2f}")
print(f"Facturación Tienda 4: ${fact_4:,.2f}")

Facturación Tienda 1: $1,150,880,400.00
Facturación Tienda 2: $1,116,343,500.00
Facturación Tienda 3: $1,098,019,600.00
Facturación Tienda 4: $1,038,375,700.00


Interpretación: Al sumar todo, me di cuenta de que la Tienda 1 es la que más plata está moviendo, superando los 1,150 millones. Por otro lado, la Tienda 4 es la que menos vende de todas. Hay una diferencia de más de 100 millones entre ellas, lo que ya nos da una pista de qué tienda podría ser la menos rentable para el señor Juan.

# 2. Ventas por categoría

In [4]:
print("Top categorías Tienda 1:\n", tienda['Categoría del Producto'].value_counts().head(3))
print("\nTop categorías Tienda 2:\n", tienda2['Categoría del Producto'].value_counts().head(3))
print("\nTop categorías Tienda 3:\n", tienda3['Categoría del Producto'].value_counts().head(3))
print("\nTop categorías Tienda 4:\n", tienda4['Categoría del Producto'].value_counts().head(3))

Top Categorías Tienda 1:
 Categoría del Producto
Muebles         465
Electrónicos    448
Juguetes        324
Name: count, dtype: int64

Top Categorías Tienda 2:
 Categoría del Producto
Muebles         442
Electrónicos    422
Juguetes        313
Name: count, dtype: int64

Top Categorías Tienda 3:
 Categoría del Producto
Muebles         499
Electrónicos    451
Juguetes        315
Name: count, dtype: int64

Top Categorías Tienda 4:
 Categoría del Producto
Muebles         480
Electrónicos    451
Juguetes        338
Name: count, dtype: int64


Interpretación: Lo que más se vende en todas las tiendas son Muebles y Electrónicos. Parece que a la gente le gusta comprar cosas grandes en Alura Store. Juguetes queda en tercer lugar en todas, así que el fuerte del negocio definitivamente no son los niños, sino el hogar y la tecnología.

# 3. Calificación promedio de la tienda


In [9]:
prom_1 = round(tienda['Calificación'].mean(), 2)
prom_2 = round(tienda2['Calificación'].mean(), 2)
prom_3 = round(tienda3['Calificación'].mean(), 2)
prom_4 = round(tienda4['Calificación'].mean(), 2)

print(f"Calificación Promedio Tienda 1: {prom_1}")
print(f"Calificación Promedio Tienda 2: {prom_2}")
print(f"Calificación Promedio Tienda 3: {prom_3}")
print(f"Calificación Promedio Tienda 4: {prom_4}")

Calificación Promedio Tienda 1: 3.98
Calificación Promedio Tienda 2: 4.04
Calificación Promedio Tienda 3: 4.05
Calificación Promedio Tienda 4: 4.0


Interpretación: Aquí los números están muy parejos, todos rondan el 4.0. Pero si nos ponemos detallistas, la Tienda 1 es la única que bajó del 4 (tiene 3.98). No es una mala nota, pero comparada con las otras, parece que los clientes de la Tienda 1 están un poquito menos conformes.

# 4. Productos más y menos vendidos

In [12]:
# Tienda 1
print(f"Tienda 1 - Más vendido: {tienda['Producto'].value_counts().idxmax()}")
print(f"Tienda 1 - Menos vendido: {tienda['Producto'].value_counts().idxmin()}")

# Tienda 2
print(f"Tienda 2 - Más vendido: {tienda2['Producto'].value_counts().idxmax()}")
print(f"Tienda 2 - Menos vendido: {tienda2['Producto'].value_counts().idxmin()}")

# Tienda 3
print(f"Tienda 3 - Más vendido: {tienda3['Producto'].value_counts().idxmax()}")
print(f"Tienda 3 - Menos vendido: {tienda3['Producto'].value_counts().idxmin()}")

# Tienda 4
print(f"Tienda 4 - Más vendido: {tienda4['Producto'].value_counts().idxmax()}")
print(f"Tienda 4 - Menos vendido: {tienda4['Producto'].value_counts().idxmin()}")


Tienda 1 - Más vendido: Microondas
Tienda 1 - Menos vendido: Auriculares con micrófono
Tienda 2 - Más vendido: Iniciando en programación
Tienda 2 - Menos vendido: Juego de mesa
Tienda 3 - Más vendido: Kit de bancas
Tienda 3 - Menos vendido: Bloques de construcción
Tienda 4 - Más vendido: Cama box
Tienda 4 - Menos vendido: Guitarra eléctrica


Interpretación: Aquí hay variedad. Mientras que en la Tienda 1 lo que manda son los Microondas, en la Tienda 4 el producto estrella es la Cama box. También es interesante ver que la "Guitarra eléctrica" es lo que menos se vende en la Tienda 4; parece que el público de esa zona no es muy musical o prefiere otros instrumentos.

# 5. Envío promedio por tienda

In [11]:
# Calculamos el promedio de la columna 'Costo de envío' y redondeamos a 2 decimales
envio_1 = round(tienda['Costo de envío'].mean(), 2)
envio_2 = round(tienda2['Costo de envío'].mean(), 2)
envio_3 = round(tienda3['Costo de envío'].mean(), 2)
envio_4 = round(tienda4['Costo de envío'].mean(), 2)

print(f"Costo de envío promedio Tienda 1: ${envio_1}")
print(f"Costo de envío promedio Tienda 2: ${envio_2}")
print(f"Costo de envío promedio Tienda 3: ${envio_3}")
print(f"Costo de envío promedio Tienda 4: ${envio_4}")

Envío Promedio Tienda 1: $26018.61
Envío Promedio Tienda 2: $25216.24
Envío Promedio Tienda 3: $24805.68
Envío Promedio Tienda 4: $23459.46


Interpretación: Los costos de envío son altos en todas, pero la diferencia es clara. La Tienda 1 es la más costosa en logística, mientras que la Tienda 4 es la más eficiente, costando unos $2,500 menos en promedio. Sin embargo, hay que ver si este ahorro compensa su baja facturación.

**Conclusión para el Sr. Juan:**

Después de ver todos los números, mi consejo es que venda la Tienda 4.

Es verdad que esta tienda tiene buenos costos de envío y buenas calificaciones, pero al final del día es la que menos dinero hace entrar a la empresa. Hay una diferencia de más de 100 millones entre lo que vende la Tienda 1 y la Tienda 4.

Aunque la Tienda 1 gasta mucho en envíos, es la que más vende, así que no conviene soltarla. Lo mejor para juntar capital es vender la sucursal que menos factura, que en este caso es la 4.
